client/payment-details/summary/index.tsx

```diff
@@ -223,19 +223,40 @@ const PaymentDetailsSummary: React.FC< PaymentDetailsSummaryProps > = ( {
 			<CardBody>
 				<div className="payment-details-summary">
 					<div className="payment-details-summary__section">
-						<p className="payment-details-summary__amount">
+						<p
+							className="payment-details-summary__amount"
+							aria-label={ __(
+								'Payment amount',
+								'woocommerce-payments'
+							) }
+						>
 							<Loadable
 								isLoading={ isLoading }
 								placeholder="Amount placeholder"
 							>
-								{ formatCurrency(
-									charge.amount,
-									charge.currency,
-									balance.currency
+								{ renderStorePrice ? (
+									<>
+										{ formatCurrency(
+											balance.amount,
+											balance.currency
+										) }
+										<span className="payment-details-summary__amount-currency">
+											{ balance.currency }
+										</span>
+									</>
+								) : (
+									<>
+										{ formatCurrency(
+											charge.amount,
+											charge.currency,
+											balance.currency
+										) }
+										<span className="payment-details-summary__amount-currency">
+											{ charge.currency }
+										</span>
+									</>
 								) }
-								<span className="payment-details-summary__amount-currency">
-									{ charge.currency || 'USD' }
```

Comment:
Nice to remove this potentially dangerous fallback to `USD`!

Reply:
I've also removed placeholder values (e.g. $0.00 USD) used for loading states in fd93486.